client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -348,6 +350,9 @@ public void setupClient(OkHttpClient.Builder builder)
                 builder.authenticator(authenticator);
                 builder.addInterceptor(authenticator);
             }
+
+            Optional<String> resolverContext = DNS_RESOLVER_CONTEXT.getValue(properties);
+            DNS_RESOLVER.getValue(properties).ifPresent(resolverClass -> builder.dns(new OkHttpDnsAdapter(instantiateDnsResolver(resolverClass, resolverContext))));
```

Comment:
Do we need the class `OkHttpDnsAdapter`? I think we could do
```java
builder.dns(instantiateDnsResolver(resolverClass, resolverContext)::lookup)

Reply:
Yes, thanks.